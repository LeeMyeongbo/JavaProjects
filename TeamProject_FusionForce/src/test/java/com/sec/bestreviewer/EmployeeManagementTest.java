package com.sec.bestreviewer;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.File;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class EmployeeManagementTest {

    @Test
    public void testArgumentsEmptyInputFile() {
        String[] args = {
                "./src/test/java/com/sec/bestreviewer/emptyInput.txt",
                "./src/test/java/com/sec/bestreviewer/emptyOutput.txt"
        };

        EmployeeManagement employeeManagement = new EmployeeManagement();
        employeeManagement.run(args);

        File file = new File("./src/test/java/com/sec/bestreviewer/emptyOutput.txt");

        assertTrue(file.exists());
    }

    @Test
    public void testArgumentsWrongArgsCount() {
        String[] args = {"input.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> employeeManagement.run(args));
    }

    @Test
    public void testArgumentsNotExistInputFile() {
        String[] args = {"notExist.txt", "emptyOutput.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> employeeManagement.run(args));
    }

    @Test
    public void integrationTest() {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_input.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);
        Approvals.verify(outputFile);
    }
}
