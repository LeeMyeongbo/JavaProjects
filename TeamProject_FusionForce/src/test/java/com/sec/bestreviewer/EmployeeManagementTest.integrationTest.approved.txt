DEL,10000000,TEST NAME,CL1,010-0000-0001,19000001,PRO
DEL,10000001,TEST NAME1,CL1,010-0000-0002,19000002,PRO
DEL,NONE
DEL,10000002,TEST NAME2,CL2,010-0000-0003,19000003,PRO
DEL,10000003,TEST NAME3,CL2,010-0000-0004,19000004,PRO
DEL,10000008,TEST NAME8,CL2,010-0000-0003,19000003,PRO
DEL,10000009,TEST NAME9,CL2,010-0000-0003,19000003,PRO
DEL,10000010,TEST NAME10,CL2,010-0000-0003,19000004,PRO
DEL,10000004,TEST NAME4,CL3,010-0000-0005,19000005,PRO
DEL,10000005,TEST NAME5,CL3,010-0000-0006,19000006,PRO
DEL,10000006,TEST NAME6,CL4,010-0000-0007,19000007,PRO
DEL,10000007,TEST NAME7,CL4,010-0000-0008,19000008,PRO
DEL,10000012,DEL AND,CL1,010-0000-0000,19001101,PRO
DEL,10000013,DEL AND1,CL2,010-0000-0001,19001104,PRO
DEL,10000014,DEL AND2,CL2,010-0000-0001,19001102,PRO
DEL,10000015,DEL AND2,CL1,010-0000-0001,19001103,PRO
MOD,10000019,MOD AND3,CL1,010-0001-0003,19001008,PRO
SCH,10000019,MOD AND3,CL1,010-0001-0003,19001118,PRO
MOD,10000016,MOD AND,CL1,010-0001-0000,19001005,PRO
MOD,10000017,MOD AND1,CL2,010-0001-0001,19001006,PRO
MOD,10000018,MOD AND2,CL2,010-0001-0002,19001007,PRO
SCH,10000016,MOD AND,CL1,010-0001-0000,19001005,EX
SCH,10000017,MOD AND1,CL2,010-0001-0001,19001006,EX
SCH,10000018,MOD AND2,CL2,010-0001-0002,19001007,EX
SCH,10000020,MOD AND4,CL3,010-0001-0004,19001008,ADV
SCH,10000021,MOD AND4,CL3,010-0001-0005,19001009,PRO
SCH,10000020,MOD AND4,CL3,010-0001-0004,19001008,ADV
SCH,10000020,MOD AND4,CL3,010-0001-0004,19001008,ADV
SCH,10000021,MOD AND4,CL3,010-0001-0005,19001009,PRO
SCH,NONE
SCH,10000022,MOD AND5,CL4,010-0002-0003,19001008,EX
MOD,10000016,MOD AND,CL1,010-0001-0000,19001005,EX
MOD,10000017,MOD AND1,CL2,010-0001-0001,19001006,EX
SCH,10000016,MOD AND,CL1,010-0001-0000,19001005,PRO
SCH,10000017,MOD AND1,CL2,010-0001-0001,19001006,PRO
SCH,10000021,MOD AND4,CL3,010-0001-0005,19001009,PRO
MOD,10000016,MOD AND,CL1,010-0001-0000,19001005,PRO
SCH,10000016,MOD AND0,CL1,010-0001-0000,19001005,PRO
SCH,10000017,MOD AND1,CL2,010-0001-0001,19001006,PRO
SCH,10000018,MOD AND2,CL2,010-0001-0002,19001007,EX
SCH,10000019,MOD AND3,CL1,010-0001-0003,19001118,PRO
SCH,10000020,MOD AND4,CL3,010-0001-0004,19001008,ADV
SCH,6
SCH,NONE
SCH,10000020,MOD AND4,CL3,010-0001-0004,19001008,ADV
SCH,NONE